### DISCUSSION TITLE
Authenticated Scans

### DISCUSSION THREAD

**Discussion by Sh4d0wHunt3rX** [blacklanternsecurity/bbot#1203] [General]
I'm not sure if BBOT currently supports this or not, that would be great if BBOT can login to the sites and get much info and links from logged in sections.

I noticed in Nuclei, they recently created "authenticated scans" feature as explained here:

https://docs.projectdiscovery.io/tools/nuclei/authenticated-scans

![image](https://github.com/blacklanternsecurity/bbot/assets/15929497/b0c7064d-65a7-429e-8a4a-e14997cb120c)

**Comment by TheTechromancer** [comment]
Interesting. I'd be curious to get @liquidsec's take on this, but I'm a bit skeptical. This seems like a huge amount of complexity to manage. The specification seems sensible, but I have a lot of questions about the code. You give it a username and password, but how does it know where the login page is? How does it know what a successful login looks like? How does it know what a valid session looks like, or when the session expires? Does it know how to get a refresh token? It's a nice idea but it seems like it would be a complete nightmare to maintain.

**Reply by Sh4d0wHunt3rX** [reply]
More info on this can be found here, maybe helpful:
https://blog.projectdiscovery.io/scanning-login-protected-targets-with-nuclei/
https://github.com/projectdiscovery/nuclei/tree/dev/pkg/authprovider/authx

Burpsuite has also authenticated scan feature, not sure how much these are related:
![image](https://github.com/blacklanternsecurity/bbot/assets/15929497/12398706-1c33-4bd9-8541-f988bb396e22)

Also, I noticed an authenticated scan feature in a irrelated service, maybe helpful:
https://www.getstark.co/blog/authenticated-url-scanning/
![image](https://github.com/blacklanternsecurity/bbot/assets/15929497/048f579c-f099-4261-9b4e-9ba0de64f8b4)

**Comment by liquidsec** [comment]
I'm also extremely skeptical of this. I have a hard time imagining a lot of use cases where adding a header (which we support now) wouldn't be enough - and that those cases would justify an extremely complicated component for managing this.
